readthedocs/core/views/hooks.py

```diff
@@ -279,8 +279,15 @@ def bitbucket_build(request):
         )
         log.debug('Bitbucket webhook payload:\n\n%s\n\n', data)
         projects = get_project_from_url(search_url)
-        if projects:
+        if projects and branches:
             return _build_url(search_url, projects, branches)
+        elif not branches:
+            log.error(
+                'Commits/Banches not found url=%s branches=%s',
+                search_url,
+                branches
+            )
+            return HttpResponseNotFound('Commits/Banches not found')
```

Comment:
There is a typo on the error message here: 'Commits/Banches' -> `Commits/branches`